src/buffer_field_api.h

```diff
@@ -0,0 +1,311 @@
+#ifndef __dispatch_buffer_field_api_h__
+#define __dispatch_buffer_field_api_h__ 1
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/** @file
+ * Inline API for common operations on buffer_fields
+ * @internal
+ * @{
+ */
+
+#include "qpid/dispatch/buffer_field.h"
+#include "qpid/dispatch/iterator.h"
+
+/* qd_buffer_field_memcpy
+ *
+ * Copy up to n octets from bfield to dest, advance bfield by the number of
+ * octets copied
+ *
+ * @return total of octets copied - may be < n if len(bfield) < n
+ */
+static inline size_t qd_buffer_field_memcpy(qd_buffer_field_t *bfield, uint8_t *dest, size_t n)
+{
```

Comment:
`if (!bfield)
     return 0;`

Reply:
Would you mind if instead we treated calling this with a null bfield as an error?  I'd like to keep these functions as fast as possible due to their use for inbound message parsing.
The proton API has done this as well by adding "assert(required-non-0-pointer)" to those APIs which expect the caller has already sanitized the parameters.  example: https://github.com/apache/qpid-proton/blob/main/c/src/core/engine.c#L99

Reply:
> Would you mind if instead we treated calling this with a null bfield as an error?
Sounds good to me.